### ISSUE TITLE
Grommet Components Sticker Sheet 1.0 - Sketch

### ISSUE THREAD

**Issue by tracybarmore** [grommet/grommet-design#66]

**Comment by gbr** [comment]
#85 is the PR associated with this issue.

**Comment by Primajin** [comment]
Does this have real textstyles and symbols set? I saw that the HP-themed ones use them, but the unstyled grommet 0.3 one does not, making it hard to use the file to actually build stuff.

![screen shot 2017-07-19 at 16 45 00](https://user-images.githubusercontent.com/1742115/28373288-4618903c-6ca2-11e7-85f1-f89335975c9b.png)

--

![screen shot 2017-07-19 at 16 46 01](https://user-images.githubusercontent.com/1742115/28373290-472d7870-6ca2-11e7-894d-bcab9b86e8ed.png)

Also it would be nice to have something like a [Craft Stylesheet](https://www.invisionapp.com/craft), so one can easily change all the colors that uses the brand color for example.

![1-whoefyv2hlm4ddngvbzw0q](https://user-images.githubusercontent.com/1742115/28373526-d9802826-6ca2-11e7-8967-d3b95dede0ea.gif)

One more question: While the website says "Applications use the "Work Sans" font." the Sketch file says "Grommet utilizes Source Sans Pro for all typography" - is this an error?

**Comment by Primajin** [comment]
Before release, please check again if all of the shared styles where correctly set.
When I change the brand color for example, I still see many elements in grommet-violet because many have them "hardcoded" instead of using the style.

![screen shot 2017-07-19 at 17 26 23](https://user-images.githubusercontent.com/1742115/28375304-6afac2e4-6ca7-11e7-9e9d-9af41e291306.png)


![screen shot 2017-07-19 at 17 26 02](https://user-images.githubusercontent.com/1742115/28375311-6e419e32-6ca7-11e7-8d32-c582a7b2b031.png)


![screen shot 2017-07-19 at 17 24 57](https://user-images.githubusercontent.com/1742115/28375315-7252b59c-6ca7-11e7-98f9-143739f18aec.png)


![screen shot 2017-07-19 at 17 23 44](https://user-images.githubusercontent.com/1742115/28375321-764d392e-6ca7-11e7-87e8-7eabc2d78548.png)

**Comment by gbr** [comment]
Thanks @Primajin for the feedback. I will make sure the file is fully consistent in regards to shared styles and post an update.

**Comment by Primajin** [comment]
Cool! Also I found that there is still some stuff from HP in there (some fonts and the second brand style)

![screen shot 2017-07-20 at 17 33 06](https://user-images.githubusercontent.com/1742115/28425972-167a3e4a-6d72-11e7-8e6e-3b40c36f9418.png)
![screen shot 2017-07-20 at 17 33 56](https://user-images.githubusercontent.com/1742115/28425971-16755858-6d72-11e7-9dba-8b137f51542b.png)

I also did some adjustments, by applying the brand style, but maybe it's better to wait for your final version...

![screen shot 2017-07-20 at 17 36 58](https://user-images.githubusercontent.com/1742115/28426002-2fe44cb8-6d72-11e7-9d03-845625e6f81f.png)

**Comment by gbr** [comment]
Hi @Primajin, please check my updated file. I created and applied shared styles to a much greater extent, as much as made sense in order to make customization of styles (and text) easier and more predictable. If you see any further issues, please do not hesitate to raise them here.

**Comment by Primajin** [comment]
Great!

![screen shot 2017-07-21 at 10 49 01](https://user-images.githubusercontent.com/1742115/28456505-d7b5e772-6e02-11e7-8c66-3a362ecac69d.png)

I still found some missing styles on the toggles, radios and range sliders though.

Also while the small video play symbol uses the `Brand stroke style` - the icon on the carousel does not use use the symbol but an own `brand video control stroke` style, to reflect the scaling and have a thickness of 8.
How about converting the borders in the small symbol to outlines, that way you can use the symbol and scale it any size you want without worrying about border thicknesses - furthermore you just need the regular `brand fill` style.

![screen shot 2017-07-21 at 11 08 22](https://user-images.githubusercontent.com/1742115/28457060-f04e27c0-6e04-11e7-9b30-e3cb2ab662fb.png)

![screen shot 2017-07-21 at 11 10 28](https://user-images.githubusercontent.com/1742115/28457138-4258586a-6e05-11e7-9788-cd185a5240f1.png)

`¯\_(ツ)_/¯`

**Comment by gbr** [comment]
That's a good suggestion. Shouldn't take too long to implement. I'm on vacation at the moment, but I'll get to updating the file as soon as I can.

**Comment by gbr** [comment]
Hi @Primajin,

Greetings! My apologies for the delay. I just got back in the office today to work on this and review my changes with the Grommet team.

We like the push towards easier customization that you've inspired. Unfortunately, we have some reservations with the Shared Styles I added in my latest push, as well as the recommendation to use outlines for the sake of a fill-only object.

We decided we should keep the Shared Styles list rather concise and agnostic to how certain niche components (buttons, particularly) are built. And who knows what components we will build in the future and whether they will add more edge cases to the Shared Styles. The list should only contain one style for "Fill" and "Stroke", separately, for each color. For that reason, we decided to remove the "Fill & Stroke" Shared Styles.

Certainly your suggestion to convert to outlines would fix that part of the issue, since outlines only have Fill. The problem with outlines is that it corrupts the pixel accuracy in regards to SVG export (something our team values). On top of that, if we did it to something like a button rectangle shape, we would lose the ability to easily round corners and make other customization. It leaves the form but destroys the blueprint, which we find useful for certain things down the road.

Finally, I should also note that the use of third-party plugins like Craft Styles are also outside our plans. We do not want to introduce additional dependencies to use our files properly (though we wouldn't inhibit people from using said plugins on their own!).

Still, your suggestions have led to a more consistent application of separate Fill and Stroke styles across all the colors. I thank you for that and will incorporate those changes in the final merge.

Note: Despite the above, I have not given up on your idea of easy global customization of colors just yet. I am going to brainstorm and research alternatives within our constraints. I am also happy to hear from you if you have any further thoughts on the matter.

Thanks and my apologies for the turn-around on this issue.
